product_pricelist_supplierinfo/models/product_pricelist_supplierinfo_mixin.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2018 Tecnativa - Vicent Cubells
+# Copyright 2018 Tecnativa - Pedro M. Baeza
+# Copyright 2019 Tecnativa - Carlos Dauden
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, tools
+
+
+class ProductPricelistSupplierinfoMixin(models.AbstractModel):
+    _name = 'product.pricelist.supplierinfo.mixin'
+
+    def _get_supplierinfo_pricelist_price(self, rule, date=None,
+                                          quantity=None):
+        """Method for getting the price from supplier info."""
+        self.ensure_one()
+        if self._name == 'product.template':
```

Comment:
This mixin is not correct if you have to put here this kind of code, and the extra model is not justified only for this. For this specific case, I would:

1. Define one new method in `product.template` with the same logic as `_get_supplierinfo_pricelist_price`, adding an extra keyword argument called `extra_domain` or similar. Add the logic for seeing if `extra_domain` has value, then sum it to the computed domain. Remove this part of code in which I'm writing over now.
2. Define `_get_supplierinfo_pricelist_price` in `product.template`, calling this method with the extra domain logic for product.template.
3. Do the same on `product.product`, but adding the extra domain for product.

Reply:
I can't add extra domain because self.id have distinct values and is assigned to diferent fields:
https://github.com/OCA/product-attribute/pull/436/files#diff-394e2468e18fe6e03a642a868c33bc75R18
https://github.com/OCA/product-attribute/pull/436/files#diff-394e2468e18fe6e03a642a868c33bc75R23

Reply:
Mixed removed